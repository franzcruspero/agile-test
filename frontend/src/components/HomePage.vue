<template>
  <div class="container">
    <div class="px-4 py-5 border shadow-lg">
      <h1 class="display-5 fw-bold">
        Values and Principles of Agile Software Development
      </h1>
      <div class="btn-group">
        <button
          type="button"
          class="pb-2 btn btn-outline-primary"
          data-bs-toggle="modal"
          data-bs-target="#values"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="16"
            height="16"
            fill="currentColor"
            class="bi bi-plus-square-fill"
            viewBox="0 0 16 16"
          >
            <path
              d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm6.5 4.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3a.5.5 0 0 1 1 0z"
            />
          </svg>
          Add Value
        </button>
        <AddModal
          type="values"
          :listReload="listReload"
          @submitForm="listReload++"
        />
        <button
          type="button"
          class="pb-2 btn btn-outline-primary"
          data-bs-toggle="modal"
          data-bs-target="#principles"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="16"
            height="16"
            fill="currentColor"
            class="bi bi-plus-square-fill"
            viewBox="0 0 16 16"
          >
            <path
              d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm6.5 4.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3a.5.5 0 0 1 1 0z"
            />
          </svg>
          Add Principle
        </button>
        <AddModal
          type="principles"
          :listReload="listReload"
          @submitForm="listReload++"
        />
      </div>
      <div class="container px-4 py-5" id="featured-3">
        <div class="row">
          <h2 class="pb-2 border-bottom">Values</h2>
        </div>
        <List :key="listReload" @updateEntry="listReload++" type="values" />
        <div class="row">
          <h2 class="pb-2 border-bottom">Principles</h2>
        </div>
        <List :key="listReload" @updateEntry="listReload++" type="principles" />
      </div>
    </div>
    <div class="b-example-divider"></div>
  </div>
</template>

<script>
import List from "@/components/List.vue";
import AddModal from "@/components/AddModal.vue";

export default {
  components: {
    List,
    AddModal,
  },
  data() {
    return {
      listReload: 0,
    };
  },
};
</script>
