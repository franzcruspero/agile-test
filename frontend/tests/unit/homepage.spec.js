import HomePage from "@/components/HomePage.vue";
import { shallowMount } from "@vue/test-utils";


describe("homePage.vue", () => {

    let wrapper = shallowMount(HomePage);

    it("shows title header of the app", () => {
        expect(wrapper.find("h1").text()).toBe("Values and Principles of Agile Software Development")
    })

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
    })
});