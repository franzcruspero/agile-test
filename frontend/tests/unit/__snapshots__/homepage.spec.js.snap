// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePage.vue renders correctly 1`] = `
<div
  class="container"
>
  <div
    class="px-4 py-5 border shadow-lg"
  >
    <h1
      class="display-5 fw-bold"
    >
       Values and Principles of Agile Software Development 
    </h1>
    <div
      class="btn-group"
    >
      <button
        class="pb-2 btn btn-outline-primary"
        data-bs-target="#values"
        data-bs-toggle="modal"
        type="button"
      >
        <svg
          class="bi bi-plus-square-fill"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm6.5 4.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3a.5.5 0 0 1 1 0z"
          />
        </svg>
         Add Value 
      </button>
      <add-modal-stub
        listreload="0"
        type="values"
      />
      <button
        class="pb-2 btn btn-outline-primary"
        data-bs-target="#principles"
        data-bs-toggle="modal"
        type="button"
      >
        <svg
          class="bi bi-plus-square-fill"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2zm6.5 4.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3a.5.5 0 0 1 1 0z"
          />
        </svg>
         Add Principle 
      </button>
      <add-modal-stub
        listreload="0"
        type="principles"
      />
    </div>
    <div
      class="container px-4 py-5"
      id="featured-3"
    >
      <div
        class="row"
      >
        <h2
          class="pb-2 border-bottom"
        >
          Values
        </h2>
      </div>
      <list-stub
        type="values"
      />
      <div
        class="row"
      >
        <h2
          class="pb-2 border-bottom"
        >
          Principles
        </h2>
      </div>
      <list-stub
        type="principles"
      />
    </div>
  </div>
  <div
    class="b-example-divider"
  />
</div>
`;
